### ISSUE TITLE
Separate section for smaller events (meetups, hackathons, etc)?

### ISSUE THREAD

**Issue by sdroege** [XAMPPRocky/timetill.rs#5]
Currently timetill.rs seems to be meant for big events like conferences/conference tracks.

Would it make sense to add a separate section for smaller things like meetups, hackathons/hackfests, etc? Currently the community calendar is used for such things but that's suboptimal for discovering events IMHO and this here looks much nicer.

Any opinions? :)

**Comment by XAMPPRocky** [comment]
Thank you for your issue! When I was designing the initial version of timetill.rs I thought about including small events, and decided against it for a few reasons.

How would timetill.rs get the information of what events are happening? Currently all the event data is stored in `conferences.json`, the current setup would be unreasonable . Smaller events data changes more frequently than conferences, venue changes, time changes, cancellations, etc. Each change would need to be a PR, adding a lot of maintenance and requiring approval before changes will be live.

Timetill.rs could instead try to fetch the information from a third party source, however each event tends to use their own service for hosting information. The most collated data source is the community calendar, however I don't believe there is an easy way to extract that information from Google Calendar.

There are further issues that come up, such as moderation. Firstly though to provide the functionality I first need an easily maintainable solution to handling the data. This isn't to say that   I'm against the idea, rather that implementing a solution that people will want to use takes a lot of thought, prototyping, and iteration.

I would be interested in hearing however what information you'd want to see a smaller event have especially anything you think services like Meetup don't provide, or give less priority?

**Comment by sdroege** [comment]
> How would timetill.rs get the information of what events are happening?

Indeed, it would either need a lot of PRs for the currently static content... or making the whole thing dynamic with some backing storage that can be updated at any time. Which increases the complexity of the whole thing considerably and then opens up questions about permissions to modify the storage, moderation, etc.

I didn't think about this part :)

>  however I don't believe there is an easy way to extract that information from Google Calendar.

AFAIU you could get caldav from Google Calendar and parse that.

> I would be interested in hearing however what information you'd want to see a smaller event have especially anything you think services like Meetup don't provide, or give less priority?

Having everything in a single place, mostly. Many things are announced on meetup but there are also lots of other places where small events are announced.

**Comment by XAMPPRocky** [comment]
> Having everything in a single place, mostly. Many things are announced on meetup but there are also lots of other places where small events are announced.

Sorry I was unclear, I was talking about information about the events themselves. What would you want to know about an event before you decide whether you're interested in attending?
